pkg/driver/mock/auto/airtemperature.go

```diff
@@ -0,0 +1,51 @@
+package auto
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"golang.org/x/exp/rand"
+
+	"github.com/smart-core-os/sc-api/go/traits"
+	"github.com/smart-core-os/sc-api/go/types"
+	"github.com/smart-core-os/sc-golang/pkg/trait/airtemperaturepb"
+	"github.com/vanti-dev/sc-bos/pkg/task/service"
+)
+
+func AirTemperatureAuto(model *airtemperaturepb.Model) *service.Service[string] {
+	slc := service.New(service.MonoApply(func(ctx context.Context, _ string) error {
+		ticker := time.NewTicker(30 * time.Second)
+		go func() {
+			randomNumber := 18 + rand.Float64()*6
+			// give each device a random set point between 18 and 24 with .05 degree accuracy
+			setPoint := math.Round(randomNumber*2) / 2
+			state := &traits.AirTemperature{
+				AmbientTemperature: &types.Temperature{
+					ValueCelsius: setPoint + (rand.Float64()*4 - 2),
+				},
+				TemperatureGoal: &traits.AirTemperature_TemperatureSetPoint{
+					TemperatureSetPoint: &types.Temperature{ValueCelsius: setPoint},
+				},
+			}
+			_, _ = model.UpdateAirTemperature(state)
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-ticker.C:
+					state.AmbientTemperature = &types.Temperature{
+						// update the ambient to be +- 2 degrees from the set point
+						ValueCelsius: setPoint + (rand.Float64()*4 - 2),
```

Comment:
While we're here, do you think you could use the live setpoint value instead of the initial set point for this randomness? I think that'd be more useful because you'd be able to change the set point via the Ops UI and the ambient temp would continue to wobble around this value. It would also mean automations and things don't get overridden.